src/cmds/net/telnetd/telnetd.c

```diff
@@ -399,30 +429,38 @@ int main(int argc, char **argv) {
 		MD(printf("Attempt to connect from address %s:%d\n",
 			inet_ntoa(client_socket.sin_addr), ntohs(client_socket.sin_port)));
 
-		for (i = 0; i < TELNETD_MAX_CONNECTIONS; i++) {
-			if (clients[i].fd == -1) {
-				break;
-			}
-		}
-
-		if (i == TELNETD_MAX_CONNECTIONS) {
+		if (client_descr >= 1000) {
+			/* there are 4 digit buffer for an accepted descriptor */
+			MD(printf("too big descriptor number=%d\n", client_descr));
 			telnet_cmd(client_descr, T_INTERRUPT, 0);
-			MD(printf("limit of connections exceded\n"));
+			close(client_descr);
 			continue;
 		}
 
-		clients[i].fd = client_descr;
-		memcpy(&clients[i].addr_in, &client_socket, sizeof(client_socket));
-
-		if (pthread_create(&thread, NULL, telnet_thread_handler, (void *) i)) {
+		child_pid = vfork();
```

Comment:
This way of creation connection handler seems unnatural for single daemon. 
I suppose pthread is not suits here? Could it be a comment in code, why?
This scheme reassembles what `inetd` does. Introducing inetd could make telnetd and others home-grown network services simplier in future. 
But not insisting on fix in this PR